### DISCUSSION TITLE
how can use strapi.entityService ?

### DISCUSSION THREAD

**Discussion by marefati110** [strapi/strapi#8115] [Legacy Topics]
i try to use ```strapi.entityService.findOne/find``` and have a problem with populate field

how should fill ```populate``` i try fill that like ```{withRelated:['MODEL']}``` but not working 

```js
const data = strapi.entityService.findOne(
      {
        params: { ...query, id },
        populate,
      },
      { model }
    );
```

DATABASE: mysql

thanks.

**Comment by derrickmehaffy** [comment]
Hi @marefati110 

The entityService isn't generally supposed to be used like that, I would recommend you use the query option that is documented here: https://strapi.io/documentation/v3.x/concepts/queries.html#findone

For popluation the 2nd parameter is an array in which you can pass in some relational fields to populate:

`strapi.query('restaurant').findOne({ id: 1 }, ['category']);`

**Reply by marefati110** [reply]
thanks for your answer but i try to overwrite fetchcall function in contentManager that use ```entityService```
so what should i do ?
